packages/contracts-bedrock/test/universal/CrossDomainMessenger.t.sol

```diff
@@ -39,6 +40,90 @@ contract CrossDomainMessenger_BaseGas_Test is CommonTest {
         assertTrue(baseGas >= minGasLimit);
     }
+
+    /// @notice Test that baseGas returns at least the floor cost for calldata
+    function test_baseGas_floor_succeeds() external view {
+        // Create a message large enough that the floor cost would be higher than the execution gas
+        bytes memory largeMessage = new bytes(100_000);
+
+        uint64 baseGasResult = l1CrossDomainMessenger.baseGas(largeMessage, 0);
+
+        // Calculate the expected floor cost
+        uint64 expectedFloorCost = l1CrossDomainMessenger.TX_BASE_GAS()
+            + (uint64(largeMessage.length) * l1CrossDomainMessenger.FLOOR_CALLDATA_OVERHEAD());
+
+        // Verify that the result is at least the floor cost
+        assertTrue(baseGasResult >= expectedFloorCost, "baseGas should return at least the floor cost");
+    }
+
+    /// @notice Test that baseGas returns the execution gas when it's higher than the floor cost
+    function test_baseGas_executionGas_succeeds() external view {
+        // Create a small message where execution gas would be higher than floor cost
+        bytes memory smallMessage = new bytes(10);
+        uint32 highGasLimit = 1_000_000;
+
+        uint64 baseGasResult = l1CrossDomainMessenger.baseGas(smallMessage, highGasLimit);
+
+        // Calculate the expected floor cost
+        uint64 floorCost = l1CrossDomainMessenger.TX_BASE_GAS()
+            + (uint64(smallMessage.length) * l1CrossDomainMessenger.FLOOR_CALLDATA_OVERHEAD());
+
+        // Calculate the expected execution gas (simplified version of what's in the contract)
+        uint64 executionGas = l1CrossDomainMessenger.RELAY_CONSTANT_OVERHEAD()
+            + l1CrossDomainMessenger.RELAY_CALL_OVERHEAD() + l1CrossDomainMessenger.RELAY_RESERVED_GAS()
+            + l1CrossDomainMessenger.RELAY_GAS_CHECK_BUFFER()
+            + (
+                (highGasLimit * l1CrossDomainMessenger.MIN_GAS_DYNAMIC_OVERHEAD_NUMERATOR())
+                    / l1CrossDomainMessenger.MIN_GAS_DYNAMIC_OVERHEAD_DENOMINATOR()
+            );
+
+        uint64 expectedExecutionGasWithOverhead = l1CrossDomainMessenger.TX_BASE_GAS() + executionGas
+            + (uint64(smallMessage.length) * l1CrossDomainMessenger.MIN_GAS_CALLDATA_OVERHEAD());
+
+        // Verify that the result is the execution gas (which should be higher than floor cost)
+        assertTrue(
+            baseGasResult >= expectedExecutionGasWithOverhead, "baseGas should return at least the execution gas"
+        );
+        assertTrue(
+            expectedExecutionGasWithOverhead > floorCost, "Execution gas should be higher than floor cost for this test"
+        );
+    }
+
+    /// @notice Fuzz test to verify the baseGas function correctly implements the Math.max logic
+    /// @param _message The message to test
+    /// @param _minGasLimit The minimum gas limit to test
+    function testFuzz_baseGas_maxLogic_succeeds(bytes calldata _message, uint32 _minGasLimit) external view {
```

Comment:
At this point we're basically just re-implementing the function. I don't know how useful tests like this really are. `testFuzz_relayMessage_baseGasSufficient_succeeds` is the only one that really matters.